### PR TITLE
Update to tagged up-rust

### PR DESCRIPTION
**Pull request by PLeVasseur**
Updates to support the [up-rust](https://crates.io/crates/up-rust/) release on crates.io

This should go in after #3 and #4

Handles #6 and #8

- [x] Realized I missed updating to use `LocalUriProvider`, will do that.

### CODE REVIEW

File: up-transport-vsomeip/examples/hello_client.rs [hunk 1]
```diff
@@ -70,11 +66,17 @@ async fn main() -> Result<(), UStatus> {
 
     // There will be a single vsomeip_transport, as there is a connection into device and a streamer
     // TODO: Add error handling if we fail to create a UPTransportVsomeip
+    let client_uuri = UUri {
```
Comment (sophokles73) [inline up-transport-vsomeip/examples/hello_client.rs:69]: you might want to use `UUri::try_from_parts` instead, which also performs some validity checks ...

File: up-transport-vsomeip/examples/hello_service.rs [hunk 2]
```diff
@@ -94,11 +89,17 @@ async fn main() -> Result<(), UStatus> {
 
     // There will be a single vsomeip_transport, as there is a connection into device and a streamer
     // TODO: Add error handling if we fail to create a UPTransportVsomeip
+    let service_uuri = UUri {
+        authority_name: HELLO_SERVICE_AUTHORITY.to_string(),
```
Comment (sophokles73) [inline up-transport-vsomeip/examples/hello_service.rs:93]: see previous comment

File: up-transport-vsomeip/src/lib.rs [hunk 3]
```diff
@@ -166,81 +183,584 @@ impl UPTransportVsomeip {
...
+        Ok(Self {
+            storage,
+            engine,
+            point_to_point_listener,
+            config_path: optional_config_path,
+            thread_handle: Some(thread_handle),
+            shutdown_runtime_tx,
+        })
+    }
+    async fn await_engine(
```
Comment (sophokles73) [inline up-transport-vsomeip/src/lib.rs:229]: maybe add a newline in between?

**Review by sophokles73 — COMMENTED** [review]
In general, I suggest to use `UUri::try_from_parts` to create UUris. You might also want to consider using the L2 API for the examples.

File: up-transport-vsomeip/src/lib.rs (hunk 3 above)
Comment (PLeVasseur) [reply up-transport-vsomeip/src/lib.rs:229]: `cargo fmt` actually removes it if I try to add it :rofl:

File: up-transport-vsomeip/examples/hello_client.rs (hunk 1 above)
Comment (PLeVasseur) [reply up-transport-vsomeip/examples/hello_client.rs]: I wasn't aware of this! Super useful.

Comment (PLeVasseur) [reply up-transport-vsomeip/examples/hello_client.rs]: Done

File: up-transport-vsomeip/examples/hello_service.rs (hunk 2 above)
Comment (PLeVasseur) [reply up-transport-vsomeip/examples/hello_service.rs]: Done

File: up-transport-vsomeip/src/lib.rs (hunk 3 above)
Comment (PLeVasseur) [reply up-transport-vsomeip/src/lib.rs:229]: Not done, due to `cargo fmt` :rofl:

**Comment by PLeVasseur** [comment]
Hey @sophokles73 -- I updated the PR with your suggestions, including adding usage of the uP-L2 APIs.

Could you take another look?

**Pull request merged**